pkg/taskrunner/tasks.go

```diff
@@ -0,0 +1,89 @@
+package taskrunner
+
+import (
+	"bytes"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"text/template"
+
+	"github.com/rs/zerolog/log"
+	"sigs.k8s.io/yaml"
+)
+
+func expandPath(d, p string) ([]string, error) {
+	return filepath.Glob(path.Join(d, p))
+}
+
+type Command interface {
+	exec(c *Config) error
+}
+
+type TaskFile struct {
+	Tasks []Task `json:"tasks"`
+}
+
+type Task struct {
+	Copy *copyTask `json:"copy,omitempty"`
+	Remove *removeTask `json:"remove,omitempty"`
+}
+
+func (t *Task) exec(c *Config) error {
+	v := reflect.ValueOf(*t)
+
+	// Ensure only one command set
+	var command Command
+	for i := 0; i < v.NumField(); i++ {
+		isNil := v.Field(i).IsNil()
+		if !isNil {
+			if command != nil {
+				return ErrConflict
+			}
+			command = v.Field(i).Interface().(Command)
+		}
+	}
+
+	return command.exec(c)
+}
```

Comment:
So `reflect` will always trigger my alarms because it incurs in penalties on code performance. What is this code trying to do? 

I see pointers above, wouldn't it be better to dereference the pointer instead?